dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.History.xml

```diff
@@ -0,0 +1,79 @@
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+-->
+
+<node>
+<!-- org.rpm.dnf.v0.History:
+   @short_description: Interface to transactions history database
+-->
+<interface name="org.rpm.dnf.v0.History">
+    <!--
+        recent_changes:
+        @options: an array of key/value pairs
+        @changeset: a dictionary {"<type_of_change>":[<package>,<package>,...]}
+
+        Get a list of packages that have been recently changed. The
+        "type_of_change" key in the output dictionary can be one of the
+        following: "installed", "removed", "upgraded", or "downgraded". The
+        packages in the output are represented as dictionaries containing the
+        requested set of attributes for the changed packages.
+
+        For installed, upgraded, and downgraded packages, the currently
+        installed version is returned.
+
+        For upgraded and downgraded packages, the "original_evr" attribute
+        is also included, containing the package EVR before the change.
+
+        If requested, upgraded packages may also include an "advisory_name"
+        field, indicating the name of the update that contains the upgraded
+        (i.e., currently installed) version of the package.
+
+        For removed packages, only the name, arch, and evr of the removed
+        version are provided.
+
+        Following options and filters are supported:
+
+            - upgraded_packages: boolean, default "true"
+                Whether to include upgraded packages in the output.
+            - downgraded_packages: boolean, default "true"
+                Whether to include downgraded packages in the output.
+            - installed_packages: boolean, default "true"
+                Whether to include installed packages in the output.
+            - removed_packages: boolean, default "true"
+                Whether to include removed packages in the output.
+            - include_advisory: boolean, default "true"
+                Whether to include "advisory_name" to upgraded packages attributes.
+            - package_attrs: list of strings, default ["name", "summary", "evr", "arch"]
+                List of package attributes that are returned for each changed
+                package. Supported attributes are the same as in list() method of
+                the Rpm interface.
```

Comment:
Would it be possible to add also `installtime` as a package attribute, which would be the latest one from the considered installations, please? It's not the same as the `buildtime`, which is already available. It could be only here, I guess the package listing cannot get the install time easily.

I just realized that the gnome-software uses that install time in its internals, in the code responsible for notifying the user about installation changes, thus it could be quite handy to have. Supposing you can get to the installation time, which I guess you can, when there's the `since` option (aka the same time as the `since` consults is fine by me).

Reply:
You mean to add "installtime" to the default value of `package_attrs`?

Actually `since` is not compared to the install time of a package, but to the rpm transaction execution time. This way we can also get the original version of an upgraded package, and also list of removed packages.

Reply:
Not to the default list necessarily, but to the list of the available package attributes (there's no such thing at the moment), while it can be available only for this history interface, not generally, especially if it's hard to get from the internal structures of the RPM/DNF. 

Hmm, is there a difference for the "added/installed/updated" packages? Not for me, but I may miss some point. For me, the transaction execution time matches install/add/update time. I do not mind package Z had been installed 2 minutes later than package A due to other 25 packages being installed before that package Z and after the transaction start. A second precision is not relevant here ;)